source/common/request_info/dynamic_metadata_impl.cc

```diff
@@ -0,0 +1,53 @@
+#include "common/request_info/dynamic_metadata_impl.h"
+
+#include "envoy/common/exception.h"
+
+namespace Envoy {
+namespace RequestInfo {
+
+DynamicMetadataImpl::~DynamicMetadataImpl() {
+  for (auto& it : data_storage_) {
+    if (it.second.destructor_) {
+      it.second.destructor_(it.second.ptr_);
+    }
+  }
+  data_storage_.clear();
+}
+
+void DynamicMetadataImpl::setDataGeneric(absl::string_view data_name, size_t type_id, void* data,
+                                         void (*destructor)(void*)) {
+  if (data_storage_.find(data_name) != data_storage_.end()) {
+    (*destructor)(data);
+    throw EnvoyException("DynamicMetadata::setData<T> called twice with same name.");
+  }
+
+  data_storage_[static_cast<std::string>(data_name)] = {type_id, data, destructor};
```

Comment:
You can static cast string view to string?

Reply:
Yep.  https://github.com/abseil/abseil-cpp/blob/master/absl/strings/string_view.h#L328